### PR TITLE
Port #1476 ('dict' option for weights_initializer and bias_initializer) to tf_legacy

### PR DESCRIPTION
**Pull request by ksbrar**
Port of https://github.com/ludwig-ai/ludwig/pull/1476/files to `tf_legacy`.

### CODE REVIEW

**Review by hungcs — APPROVED** [review]

**Comment by ksbrar** [comment]
@justinxzhao Should I add the pre-commit.yaml file to `tf-legacy` as part of this PR? (it is not present otherwise)

File: ludwig/combiners/combiners.py [hunk 1]
```diff
@@ -60,8 +60,8 @@ class ConcatCombinerConfig:
     num_fc_layers: int = schema.NonNegativeInteger(default=0)
     fc_size: int = schema.PositiveInteger(default=256)
     use_bias: bool = True
-    weights_initializer: str = schema.InitializerOptions(default='glorot_uniform')
-    bias_initializer: str = schema.InitializerOptions(default='zeros')
+    weights_initializer: Union[str, Dict] = schema.InitializerOrDict(default='glorot_uniform')
+    bias_initializer: Union[str, Dict] = schema.InitializerOrDict(default='zeros')
```
Comment (justinxzhao) [inline ludwig/combiners/combiners.py:64]: Out of curiosity, what's an example of a bias initializer configured using a `dict`?

**Comment by justinxzhao** [comment]
> @justinxzhao Should I add the pre-commit.yaml file to `tf-legacy` as part of this PR? (it is not present otherwise)

Feel free to ignore the pre-commit check in this branch, since we've never fixed all of the linting/formatting errors on `tf-legacy` and the precommit check will always be red. I'll see if there's a way to disable precommit checking for PRs that aren't merging into `master`.

File: ludwig/combiners/combiners.py (hunk 1 above)
Comment (ksbrar) [reply ludwig/combiners/combiners.py:64]: Don't have a concrete example for you 😅 but it's explicity supported by the [developer guide](https://ludwig-ai.github.io/ludwig-docs/user_guide/configuration/combiners/#concat-combiner), traces to [TF documentation for initializer parameters](https://www.tensorflow.org/api_docs/python/tf/keras/initializers)

**Review by tgaddair — APPROVED** [review]

**Pull request merged**